File: app/src/main/java/com/funckyhacker/fileexplorer/view/main/MainViewModel.java
package com.funckyhacker.fileexplorer.view.main;

import android.content.ContentResolver;
import android.databinding.BaseObservable;
import android.databinding.Bindable;
import android.net.Uri;
import android.support.annotation.NonNull;
import com.funckyhacker.fileexplorer.view.adapter.MainGridAdapter;
import com.funckyhacker.fileexplorer.view.adapter.MainLinearAdapter;
import java.io.File;
import java.util.List;

public abstract class MainViewModel extends BaseObservable {

  public abstract void init(MainView view);

  public abstract void setData(List<File> files);

  @Bindable public abstract String getCurrentPath();

  public abstract int getPageSize();

  public abstract void setCurrentPath(String currentPath);

  public abstract void setFilesToList(@NonNull String name);

  public abstract void setFilesToList(@NonNull File file);

  public abstract void popItem();

  @Bindable public abstract boolean isNoFiles();

  public abstract void sendIntent(ContentResolver resolver, File file, Uri uri);

  public abstract int getLayoutType();

  public abstract void setLayoutType(int type);

  public abstract boolean isSearchMode();

  public abstract void setSearchMode(boolean isSearchMode);

  public abstract MainLinearAdapter getLinearAdapter();

  public abstract MainGridAdapter getGridAdapter();

}

File: app/src/main/java/com/funckyhacker/fileexplorer/view/main/MainViewModel.kt
package com.funckyhacker.fileexplorer.view.main

import android.content.ContentResolver
import android.databinding.BaseObservable
import android.databinding.Bindable
import android.net.Uri
import com.funckyhacker.fileexplorer.view.adapter.MainGridAdapter
import com.funckyhacker.fileexplorer.view.adapter.MainLinearAdapter
import java.io.File

abstract class MainViewModel : BaseObservable() {

    @get:Bindable
    abstract var currentPath: String

    abstract val pageSize: Int

    @get:Bindable
    abstract val isNoFiles: Boolean

    abstract var layoutType: Int

    abstract var isSearchMode: Boolean

    abstract val linearAdapter: MainLinearAdapter

    abstract val gridAdapter: MainGridAdapter

    abstract fun init(view: MainView)

    abstract fun setData(files: List<File>?)

    abstract fun setFilesToList(name: String)

    abstract fun setFilesToList(file: File)

    abstract fun popItem()

    abstract fun sendIntent(resolver: ContentResolver, file: File, uri: Uri)

}
